### PR TITLE
adding json generation support

### PR DESCRIPTION
**Pull request by hermankailey**
adding json generation and user feature install support for #15523

### CODE REVIEW

**Comment by hermankailey** [comment]
#build

**Comment by hermankailey** [comment]
#build

**Comment by hermankailey** [comment]
#build

File: dev/com.ibm.ws.install.featureUtility/resources/com/ibm/ws/install/featureUtility/internal/resources/FeatureUtilityToolOptions.nlsprops [hunk 1]
```diff
@@ -56,6 +56,14 @@ installFeature.option-desc.--to=\
 \tthis option is not specified, the feature is installed as a user     \n\
 \tfeature.
 #------------------------------\n at 72 chars -- leading tab-----------\n\#
+installFeature.option-key.--featuresBom=\
+\ \ \ \ --featuresBom=Maven coordinate
+installFeature.option-desc.--featuresBom=\
+\tSpecify the Maven coordinate for the features BOM file. This option  \n\
+\tallows for the installation of user features from the configured     \n\
+\tmaven repository. Ensure that the repository also contains the       \n\
+\tassociated features JSON file within the same groupID as the BOM.
```
Comment (Charlotte-Holt) [inline dev/com.ibm.ws.install.featureUtility/resources/com/ibm/ws/install/featureUtility/internal/resources/FeatureUtilityToolOptions.nlsprops]: Suggested update

>This option allows for the installation of user features from the configured maven repository. Ensure that the repository also contains the associated features JSON file within the same groupID as the BOM.

To

>This option allows for the installation of user features from the configured **M**aven repository. Ensure that the repository also contains the associated features JSON file **in** the same **groupId** as the BOM **file**.

File: dev/com.ibm.ws.install/resources/com/ibm/ws/install/internal/resources/InstallKernel.nlsprops [hunk 2]
```diff
@@ -886,4 +886,8 @@ MAPBASED_ERROR_UNSUPPORTED_FILE=CWWKF1502E: The {0} file cannot be installed bec
 MAPBASED_ERROR_UNSUPPORTED_FILE.explanation=The file was not installed because an unsupported type of file was provided.
 MAPBASED_ERROR_UNSUPPORTED_FILE.useraction=See the problem determination information on the WebSphere Application Server Support web page: http://www.ibm.com/software/webservers/appserv/was/support.
 
+ERROR_INVALID_FEATURE_BOM_COORDINATE=CWWKF1503E: The format of the {0} features BOM Maven coordinate is invalid. Ensure that the coordinate is in the groupId:artifactId:version format.
```
Comment (Charlotte-Holt) [inline dev/com.ibm.ws.install/resources/com/ibm/ws/install/internal/resources/InstallKernel.nlsprops]: Suggested update

>The format of the {0} features BOM Maven coordinate **isn't** valid. Ensure that the coordinate is in the groupId:artifactId:version format.

File: dev/com.ibm.ws.install/resources/com/ibm/ws/install/internal/resources/InstallKernel.nlsprops [hunk 3]
```diff
@@ -886,4 +886,8 @@ MAPBASED_ERROR_UNSUPPORTED_FILE=CWWKF1502E: The {0} file cannot be installed bec
...
+ERROR_INVALID_FEATURE_BOM_COORDINATE.explanation=The features BOM Maven coordinate format is invalid.
```
Comment (Charlotte-Holt) [inline dev/com.ibm.ws.install/resources/com/ibm/ws/install/internal/resources/InstallKernel.nlsprops]: Suggested update

>The features BOM Maven coordinate format **isn't** valid.

**Review by Charlotte-Holt — COMMENTED** [review]
@herman-kailey I added ID review. Let me know if you have any questions!

File: dev/com.ibm.ws.install.featureUtility/resources/com/ibm/ws/install/featureUtility/internal/resources/FeatureUtilityToolOptions.nlsprops (hunk 1 above)
Comment (hermankailey) [reply dev/com.ibm.ws.install.featureUtility/resources/com/ibm/ws/install/featureUtility/internal/resources/FeatureUtilityToolOptions.nlsprops]: fixed

**Comment by hermankailey** [comment]
#build

**Comment by hermankailey** [comment]
#build

**Comment by jjiwooLim** [comment]
#libbybot

**Comment by skasund** [comment]
L2 message review completed

**Comment by jjiwooLim** [comment]
#build

**Comment by jjiwooLim** [comment]
IM personal build:
https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.build.viewResult&id=_P-VgQLRcEeuLuaSPi55stQ

**Comment by jjiwooLim** [comment]
#libbybot

**Review by bensonlatibm — APPROVED** [review]
looks good!

**Pull request merged**